### PR TITLE
[bcg729] New port

### LINKED ISSUE #14835: [New Port Request] bcg729
Library name: bcg729

Library description: G.729 speech codec

Source repository URL: https://github.com/BelledonneCommunications/bcg729

Project homepage (if different from the source repository): https://linphone.org/technical-corner/bcg729

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

Can be built with CMake.

### PR DESCRIPTION
**Pull request by geraldcombs**
**Describe the pull request**

This adds a port for Belledonne Communications' G.729 codec.

- What does your PR fix? Fixes #14835

- Which triplets are supported/not supported?

Tested with x64-osx, but should work with *-windows and *-linux.

- Have you updated the CI baseline?

Yes

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -33,6 +33,8 @@ apr:arm64-windows=fail
 # Requires ATL for ARM64 to be installed in CI
 azure-storage-cpp:arm64-windows=fail
 
+bcg729:arm-uwp=fail
+bcg729:x64-uwp=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:37]: Why do you set these two triplets fail? Does this port not support `uwp`?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: scripts/ci.baseline.txt (hunk 1 above)
Comment (geraldcombs) [reply scripts/ci.baseline.txt]: Confusion on my part from the initial CI failures. I've removed the entries.

File: ports/bcg729/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    HEAD_REF master
+)
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    set(ENABLE_STATIC ON)
+    set(ENABLE_SHARED OFF)
+else()
+    set(ENABLE_STATIC OFF)
+    set(ENABLE_SHARED ON)
+endif()
```
Comment (NancyLi1013) [inline ports/bcg729/portfile.cmake:15]:
```suggestion
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" ENABLE_STATIC )
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" ENABLE_SHARED)
```
Please see this doc https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#choose-either-static-or-shared-binaries

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR. @geraldcombs

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**